lib/source-map.js

```diff
@@ -74,7 +68,12 @@ class SourceMap {
 
 
   addFile(filename) {
-    let source = ensurePosix(fs.readFileSync(this._resolveFile(filename), 'utf-8'));
+    let source = '';
+    if (path.isAbsolute(filename)) {
+      source = ensurePosix(fs.readFileSync(filename));
```

Comment:
Let's consider moving this logic to the input abstraction, then this library merely needs to support a pluggable fs.